bin/node/cli/src/service.rs

```diff
@@ -96,8 +96,22 @@ macro_rules! new_full_start {
 				import_setup = Some((block_import, grandpa_link, babe_link));
 				Ok(import_queue)
 			})?
-			.with_rpc_extensions(|client, pool, _backend, fetcher, _remote_blockchain| -> Result<RpcExtension, _> {
-				Ok(node_rpc::create(client, pool, node_rpc::LightDeps::none(fetcher)))
+			.with_rpc_extensions(|client, pool, _backend, _fetcher, _remote_blockchain, select_chain, keystore| ->
+				Result<RpcExtension, _>
+			{
+				let babe_link = import_setup.as_ref().map(|s| &s.2)
+					.expect("Link is present or set up failed; qed.");
```

Comment:
This isn't really a proof, why is it an `Option` in the first place? The list of deps here is getting out of hand too.

Reply:
It was used for getting instances out of the `ServiceBuilder::with_import_queue` callback.

https://github.com/paritytech/substrate/blob/b27f178f06266e9d10bbb0d80f566b7beff171fc/bin/node/cli/src/service.rs#L96

I just copied the proof from here:
https://github.com/paritytech/substrate/blob/b27f178f06266e9d10bbb0d80f566b7beff171fc/bin/node/cli/src/service.rs#L158

Reply:
Yeah, that's what I was expecting, although the original proof has way more info than yours, it mentions that it's the `full` node that have them set, yours just basically state: "it's there or something is wrong", but doesn't mention why it should be there in the first place.